pkg/object/mem.go

```diff
@@ -129,16 +129,40 @@ func (m *memStore) Delete(key string) error {
 }
 
 func (m *memStore) List(prefix, marker, delimiter string, limit int64) ([]Object, error) {
-	if delimiter != "" {
-		return nil, notSupportedDelimiter
-	}
 	m.Lock()
 	defer m.Unlock()
 
 	objs := make([]Object, 0)
+	commonPrefixsMap := make(map[string]bool, 0)
 	for k := range m.objects {
 		if strings.HasPrefix(k, prefix) && k > marker {
 			o := m.objects[k]
+			if delimiter != "" {
+				remainString := strings.TrimPrefix(k, prefix)
+				pos := strings.Index(remainString, delimiter)
```

Comment:
The range of pos is `[-1,len(remainString)-1]`, so `pos! = len(remainString)` is not necessary.

Reply:
if pos != -1, but the pos == len(remainString), I don't want to handle this situation, it is considered a complete object and will be handled by the following logic.

Reply:
Sorry, I think I know, I should write pos != len(remainString) -1 here, I'll fix him

Reply:
fixed

Reply:
```
remainString=ab/ 
delimiter=/
pos=2
```
This case should in `commonPrefix`,so just remove `&& pos != (len(remainString)-1)`

Reply:
This is a full object that does not belong to commonPrefix. It will proceed to the following logic and return as a full object.

Reply:
The full object  as commonPrefix is ok.
this is a example :
aliyun oss
<img width="1968" alt="image" src="https://user-images.githubusercontent.com/31313340/227866437-0ef46305-71a3-489e-a314-c4f75c7e486b.png">

Reply:
Sure,  I also return this full object here, just instead of going to commonPrefix logic to handle it, but as an object, which can fill more real object properties.
![image](https://user-images.githubusercontent.com/8816403/227868333-62f35029-2a5c-43b0-ab82-f0f1e761922a.png)

Reply:
It is best to use the same logic as other object stores for handling commprefix. The commprefix can also have more information, albeit with a zero value.
```
if delimiter != "" {
    remainString := strings.TrimPrefix(k, prefix)
    if pos := strings.Index(remainString, delimiter);pos != -1 {
	    commonPrefix := remainString[0 : pos+1]
	    if _, ok := commonPrefixsMap[commonPrefix]; ok {
		    continue
	    }
	    f := &file{
		    obj{
			    prefix + commonPrefix,
			    0,
			    time.Unix(0, 0),
			    strings.HasSuffix(commonPrefix, "/"),
		    },
		    o.owner,
		    o.group,
		    o.mode,
		    false,
	    }
	    objs = append(objs, f)
	    commonPrefixsMap[commonPrefix] = true
	    continue
    }
}
```

Reply:
Ok, I fix it.